### PR TITLE
Twap Fix Spot Price Quote/Base

### PR DESCRIPTION
**Pull request by stackman27**
Closes: #XXX

## What is the purpose of the change

We had an issue where PoolManager Base/Spot price was inaccurate i.e. denom0 = base, denom1 = qupte 
which we fixed here: https://github.com/osmosis-labs/osmosis/pull/5863/files#diff-9ff53a5917532eb5c22f23f3718d45ff3b7d738df6f17652b3af5b39a30d583eR111

however, twap denoms were always accurate denom0 = quote, denom1 = base 
as you can see here (https://github.com/osmosis-labs/osmosis/blob/main/x/twap/logic.go#L49-L52) 

because of the inaccuracy in poolManager spot price SP0 and SP1 of twap was inaccurate, 
therefore we only need to flip the spot price of twap record and no need to flip the denoms

### CODE REVIEW

**Comment by stackman27** [comment]
@devbot-wizard help

**Comment by devbot-wizard** [comment]
Hi! I'm DevBot, a bot that helps with common tasks in the development process. Commands:
- `devbot fix`: Fix basic errors in the PR. (e.g. imports, changelog conflicts, go mod conflicts)
- `devbot merge base`: Merge the base branch into the PR branch.
- `devbot add changelog [misc/feat/bug/api/sec] [message]`: Add a changelog entry to the PR. (e.g. `devbot add changelog feat Added a new feature`)
    - If message is blank, defaults to PR title.
- `devbot re-pr`: Re-PR the PR, useful for external contributors where we have no edit perms.

**Review by czarcas7ic — APPROVED** [review]
This change makes sense, unfortunate that we looked over it in prior review.

File: app/upgrades/v17/upgrades_test.go [hunk 1]
```diff
@@ -72,8 +72,6 @@ func dummyTwapRecord(poolId uint64, t time.Time, asset0 string, asset1 string, s
 }
 
 func assertTwapFlipped(suite *UpgradeTestSuite, pre, post types.TwapRecord) {
-	suite.Require().Equal(pre.Asset0Denom, post.Asset1Denom)
-	suite.Require().Equal(pre.Asset1Denom, post.Asset0Denom)
```
Comment (mattverse) [inline app/upgrades/v17/upgrades_test.go:76]: Can we still have checks for preAsset0Denom and post.Asset0Denom, same for denom1?

**Review by mattverse — APPROVED** [review]
ACK

**Comment by stackman27** [comment]
devbot merge base

**Comment by stackman27** [comment]
@devbot-wizard merge base

**Comment by stackman27** [comment]
mergining since we have 2 approvals

**Pull request merged**